shepherd_bms/src/analyzer.cpp

```diff
@@ -274,4 +273,29 @@ void Analyzer::calcStateOfCharge()
         float distance_from_higher = (bmsdata->minVoltage.val) - ((index / 10) + 2.9);
         bmsdata->soc = ((distance_from_higher*STATE_OF_CHARGE_CURVE[index+1]) + ((1-distance_from_higher)*STATE_OF_CHARGE_CURVE[index]));
     }
+}
+
+void Analyzer::calcPackResistances(AccumulatorData_t *bms_data)
+{
+    float packResistances[NUM_CHIPS][NUM_CELLS_PER_CHIP];
+    if(bms_data->packCurrent >= 100)
+    {
+        for(int i = 0; i < NUM_CHIPS; i++)
+        {
+            for(int j = 0; j < NUM_CELLS_PER_CHIP; j++)
+            {
+                packResistances[i][j] = (bms_data->chipData[i].openCellVoltage[j] - bms_data->chipData[i].voltageReading[j]) / bms_data->packCurrent;
+            }
+        };
+    }
+    if(bms_data->packCurrent < 5)
+    {
+       for(int i = 0; i < NUM_CHIPS; i++)
```

Comment:
This is a comment for @dyldonahue but we should 110% look into some sort of unit testing framework people can test out their code logic in beforehand, I'd like to start seeing unit tests in addition to code in the future, but that might be an unreasonable ask for code on an EE team lmao

Reply:
I agree and have been trying to look into it, I've kinda come to the conclusion that it just might not be doable for 17 and working in the short term. If we got some people with some experience there maybe, but otherwise I'm not sure the timeline checks out for me to learn some things that I need to to get it there right now